maintenance/kubernetes/aks-1.22.md

```diff
@@ -0,0 +1,192 @@
+
+  # upgrade control-plane first
+  $ az aks upgrade \
+      --resource-group prod-jenkins-private-prod \
+      --name temp-privatek8s \
+      --control-plane-only \
+      --kubernetes-version 1.22
+
+      Kubernetes may be unavailable during cluster upgrades.
+      Are you sure you want to perform this operation? (y/N): y
+     Since control-plane-only argument is specified, this will upgrade only the control plane to 1.22. Node pool will not change. Continue? (y/N): y
+
+  $ az aks show --resource-group prod-jenkins-private-prod --name temp-privatek8s --output table
+  Name             Location    ResourceGroup              KubernetesVersion    CurrentKubernetesVersion    ProvisioningState    Fqdn
+  ---------------  ----------  -------------------------  -------------------  --------------------------  -------------------  --------------------------------------------------
+  temp-privatek8s  eastus2     prod-jenkins-private-prod  1.22                 1.22.6                      Upgrading            temp-privatek8s-dns-c5f4426a.hcp.eastus2.azmk8s.io
+
+  $ az aks show --resource-group prod-jenkins-private-prod --name temp-privatek8s --output table
+  Name             Location    ResourceGroup              KubernetesVersion    CurrentKubernetesVersion    ProvisioningState    Fqdn
+  ---------------  ----------  -------------------------  -------------------  --------------------------  -------------------  --------------------------------------------------
+  temp-privatek8s  eastus2     prod-jenkins-private-prod  1.22                 1.22.6                      Succeeded            temp-privatek8s-dns-c5f4426a.hcp.eastus2.azmk8s.io
+
+  # upgrade the nodes
+  $ az aks upgrade \
+      --resource-group prod-jenkins-private-prod \
+      --name temp-privatek8s \
+      --kubernetes-version 1.22
+
+      Kubernetes may be unavailable during cluster upgrades.
+      Are you sure you want to perform this operation? (y/N): y
+     The cluster is already on version 1.22 and is not in a failed state. No operations will occur when upgrading to the same version if the cluster is not in a failed state.
+     Since control-plane-only argument is not specified, this will upgrade the control plane AND all nodepools to version 1.22. Continue? (y/N): y
+
+  $ az aks show --resource-group prod-jenkins-private-prod --name temp-privatek8s --output table
+  Name             Location    ResourceGroup              KubernetesVersion    CurrentKubernetesVersion    ProvisioningState    Fqdn
+  ---------------  ----------  -------------------------  -------------------  --------------------------  -------------------  --------------------------------------------------
+  temp-privatek8s  eastus2     prod-jenkins-private-prod  1.22                 1.22.6                      Succeeded            temp-privatek8s-dns-c5f4426a.hcp.eastus2.azmk8s.io
+```
+
+    - [x] Starting the upgrade in Azure (prodpublick8s)
```

Comment:
Not sure why this one is rendered as a code block in the PR preview... Not really important.

Reply:
indentation ... one less and it's ok